asyncpg/protocol/protocol.pyx

```diff
@@ -725,6 +725,21 @@ cdef class BaseProtocol(CoreProtocol):
             self.last_query = None
             self.return_extra = False
 
+    cdef _on_notice(self, parsed):
+        # Check it here to avoid unnecessary `_field_map` iteration
+        cb = self.connection._notice_cb_set
+        if cb:
+            fields = apg_exc_base.PostgresMessage._field_map
+            message = {}
+            for k, v in parsed.items():
+                try:
+                    message[fields[k]] = v
+                except KeyError:
+                    # Ignore new unknown fields
+                    pass
+
+            self.connection._notice(message)
```

Comment:
A simple class is not much "heavier" than a dict.  Furthermore, notices should be a rare occurrence (compared to Notify).  I don't really think that the ability to enumerate fields is all that useful.  On the other hand, if notices are instances of `PostgresNotice` object, we can make its `__str__` default to `message` and provide access to the message dict via the `as_dict()` method.  IMO, it's a much nicer interface than a plain dict.